### PR TITLE
Add specific intents for item consumption and preferences

### LINKED ISSUE #111: Add specific intents for item consumption and preferences
This is when the user wants to tell the agent that it has already consumed that item (e.g., seen a movie) or express a like/dislike preference about an item recommendation.

It needs to be declared in interaction_model.py, but also needs a corresponding templated response.

More generally, a list of minimum intents should be defined and initialized in __init__ and checked that these are provided in the interaction model yaml file.

### PR DESCRIPTION
**Pull request by NoB0**
Add a list of minimum required intents that need to be defined in the YAML configuration file. 

Fixes #111

### CODE REVIEW

File: usersimcrs/simulator/agenda_based/interaction_model.py [hunk 1]
```diff
@@ -61,6 +57,23 @@ def __init__(
         # Keep track of the current user intent.
         self._current_intent = self._agenda.pop()
 
+    def _initialize_required_intent(self) -> None:
```
Comment (kbalog) [inline usersimcrs/simulator/agenda_based/interaction_model.py:60]: Nit: `_initialize_required_intents`

File: usersimcrs/simulator/agenda_based/interaction_model.py [hunk 2]
```diff
@@ -61,6 +57,23 @@ def __init__(
...
+        """Initializes required intent.
```
Comment (kbalog) [inline usersimcrs/simulator/agenda_based/interaction_model.py:61]: Nit: intents (in plural)

File: usersimcrs/simulator/agenda_based/interaction_model.py [hunk 3]
```diff
@@ -61,6 +57,23 @@ def __init__(
...
+
+        Raises:
+            RuntimeError: if some required intents are not defined.
+        """
+        required_intents = self._config.get("required_intents", {})
+        if not self.REQUIRED_INTENTS.issubset(required_intents.keys()):
+            raise RuntimeError(
+                f'The interaction model {self._config.get("name")} needs to '
+                "defined the following intents under required_intents: "
```
Comment (kbalog) [inline usersimcrs/simulator/agenda_based/interaction_model.py:70]: Nit: "define the ..." (or "have the following intents defined ...")

File: usersimcrs/simulator/agenda_based/interaction_model.py [hunk 4]
```diff
@@ -16,22 +16,16 @@
...
-    INTENT_ITEM_CONSUMED = Intent("NOTE.YES")
-    # The user responds to a recommendation positively or negatively.
-    INTENT_LIKE = Intent("NOTE.LIKE")
-    INTENT_DISLIKE = Intent("NOTE.DISLIKE")
-    INTENT_NEUTRAL = Intent("ADD-INTENT-NAME")  # TODO: intent to be created
-    # The user expresses preference on some slot/value.
-    INTENT_DISCLOSE = Intent("DISCLOSE")
-    # The user cannot answer.
-    INTENT_DONT_KNOW = Intent("ADD-INTENT-NAME")  # TODO: intent to be created
+    REQUIRED_INTENTS = {
```
Comment (kbalog) [inline usersimcrs/simulator/agenda_based/interaction_model.py:19]: Can you add a comment explaining what this is for?

**Review by kbalog — APPROVED** [review]
LGTM

We discussed revising and possibly simplifying the CRS_v1 intent scheme? Can you please create an issue for that?

File: usersimcrs/simulator/agenda_based/interaction_model.py (hunk 1 above)
Comment (NoB0) [reply usersimcrs/simulator/agenda_based/interaction_model.py:60]: Done.

File: usersimcrs/simulator/agenda_based/interaction_model.py (hunk 2 above)
Comment (NoB0) [reply usersimcrs/simulator/agenda_based/interaction_model.py:61]: Done.

File: usersimcrs/simulator/agenda_based/interaction_model.py (hunk 4 above)
Comment (NoB0) [reply usersimcrs/simulator/agenda_based/interaction_model.py:19]: Done.

**Comment by NoB0** [comment]
I created a follow-up issue regarding the revision of CRS_v1 intent scheme. #131

**Pull request merged**